odl/tomo/geometry/geometry.py

```diff
@@ -155,83 +173,189 @@ def translation(self):
+            If ``mparam`` is a single parameter, a single vector is
+            returned, otherwise a stack of vectors along axis 0.
         """
         raise NotImplementedError('abstract method')
 
-    def rotation_matrix(self, mpar):
-        """Return the rotation matrix from initial state to state at ``mpar``.
+    def rotation_matrix(self, mparam):
+        """Return the rotation matrix to the system state at ``mparam``.
 
         Parameters
         ----------
-        mpar : `motion_params` element
-            Motion parameter for which to calculate the rotation matrix.
+        mparam : `array-like`
+            Motion parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
 
         Returns
         -------
-        rot : `numpy.ndarray`, shape (`ndim`, `ndim`)
-            The rotation matrix mapping vectors at the initial state
-            to the ones in the state defined by ``mpar``. The rotation
-            is extrinsic, i.e., defined in the fixed ("world") coordinate
-            system.
+        rot : `numpy.ndarray`, shape (ndim, ndim) or (num_mparams, ndim, ndim)
+            The rotation matrix (or matrices) mapping vectors at the
+            initial state to the ones in the state defined by ``mparam``.
+            The rotation is extrinsic, i.e., defined in the "world"
+            coordinate system.
+            If ``mparam`` is a single parameter, a single matrix is
+            returned, otherwise a stack of matrices along axis 0.
         """
         raise NotImplementedError('abstract method')
 
-    def det_to_src(self, mpar, dpar, normalized=True):
+    def det_to_src(self, mparam, dparam, normalized=True):
         """Vector pointing from a detector location to the source.
 
-        A function of the motion and detector parameters.
-
         Parameters
         ----------
-        mpar : `motion_params` element
-            Motion parameter at which to evaluate.
-        dpar : `det_params` element
-            Detector parameter at which to evaluate.
+        mparam : `motion_params` element or `array-like`
+            Motion parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
+        dparam : `det_params` element or `array-like`
+            Detector parameter(s) at which to evaluate. An array should
+            stack parameters along axis 0.
         normalized : bool, optional
-            If ``True``, return a normalized (unit) vector.
+            If ``True``, normalize the resulting vector(s) to unit length.
 
         Returns
         -------
-        vec : `numpy.ndarray`, shape (`ndim`,)
-            (Unit) vector pointing from the detector to the source
+        vec : `numpy.ndarray`
+            (Unit) vector(s) pointing from the detector to the source.
+            If both ``mparam`` and ``dparam`` are single parameters, a single
+            vector is returned, otherwise a stack of vectors.
```

Comment:
Turned out that for the "minor tweaks" I had to rethink the whole vectorization implementation, but now I'm pretty happy with what it supports.
Now you can use broadcasting across all parameters (like 2 angles and 2 detector parameters), and for that the parameters must be provided one component at a time, not one parameter at a time (otherwise no broadcasting possible).

So now it is like this (in the extreme case):
```python
        >>> geom = odl.tomo.Parallel3dEulerGeometry(...)
        >>> # 2 values for each variable, resulting in 2 vectors
        >>> angles = ([0, np.pi / 2], [0, np.pi])
        >>> dparams = ([-1, 0], [-1, 0])
        >>> pts = geom.det_point_position(angles, dparams)
        >>> pts[0]  # Corresponds to angle = (0, 0), dparam = (-1, -1)
        array([-1.,  1., -1.])
        >>> pts.shape
        (2, 3)
        >>> # 4 x 5 parameters for both
        >>> angles = dparams = (np.zeros((4, 5)), np.zeros((4, 5)))
        >>> geom.det_point_position(angles, dparams).shape
        (4, 5, 3)
        >>> # Broadcasting angles to shape (4, 5, 1, 1)
        >>> angles = (np.zeros((4, 1, 1, 1)), np.zeros((1, 5, 1, 1)))
        >>> # Broadcasting dparams to shape (1, 1, 6, 7)
        >>> dparams = (np.zeros((1, 1, 6, 1)), np.zeros((1, 1, 1, 7)))
        >>> # Total broadcast parameter shape is (4, 5, 6, 7)
        >>> geom.det_point_position(angles, dparams).shape
        (4, 5, 6, 7, 3)
```